sdk/storage/Azure.Storage.Files.DataLake/src/Models/GetFileSystemsAsyncCollection.cs

```diff
@@ -28,26 +28,165 @@ internal class GetFileSystemsAsyncCollection : StorageCollectionEnumerator<FileS
             _prefix = prefix;
         }
 
-        public override async ValueTask<Page<FileSystemItem>> GetNextPageAsync(
-            string continuationToken,
-            int? pageSizeHint,
-            bool isAsync,
-            CancellationToken cancellationToken)
+        private static Page<FileSystemItem> ConvertPage(Page<BlobContainerItem> page)
         {
-            Task<Response<BlobContainersSegment>> task = _client.GetBlobContainersInternal(
-                continuationToken,
-                (BlobContainerTraits)_traits,
-                _prefix,
-                pageSizeHint,
-                isAsync,
-                cancellationToken);
-            Response<BlobContainersSegment> response = isAsync ?
-                await task.ConfigureAwait(false) :
-                task.EnsureCompleted();
             return Page<FileSystemItem>.FromValues(
-                response.Value.BlobContainerItems.Select(item => item.ToFileSystemItem()).ToArray(),
-                response.Value.NextMarker,
-                response.GetRawResponse());
+                page.Values.Select(item => item.ToFileSystemItem()).ToArray(),
```

Comment:
Nit - just pass `ConvertItem` instead of a lambda so we've got one spot to change in the future if needed.